internal/gitinterface/utils.go

```diff
@@ -37,3 +38,35 @@ func RemoteRef(refName, remoteName string) string {
 
 	return remotePath
 }
+
+// IsNiceGitVersion determines whether the version of git is "nice". Certain Git
+// subcommands that gittuf uses were added in newer versions than some common
+// client versions. Instead of using a workaround for all clients, we determine
+// if we can use the newer features or instead need to use workarounds.
+func IsNiceGitVersion() (bool, error) {
+	cmd := exec.Command("git", "--version")
+	output, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+
+	versionString := strings.TrimPrefix(strings.TrimSpace(string(output)), "git version ")
+	var major, minor, patch int
+
+	_, err = fmt.Sscanf(versionString, "%d.%d.%d", &major, &minor, &patch)
+	if err != nil {
+		return false, err
+	}
+
+	switch {
+	case major >= 2:
+		switch {
+		case minor >= 38:
+			return true, nil
+		default:
+			return false, nil
+		}
+	default:
+		return false, nil
+	}
```

Comment:
Could this instead be a if statment like this?

``` go
if major >= 2 && minor >= 38 {
    return true, nil
}  
return false, nil
```

Reply:
Prefer the if statements as well.